From 3e8bf88153fa12c6117a068cba157da29506f3af Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Fri, 20 May 2022 09:45:56 +0300
Subject: [PATCH] [SW-0] common: add wfe for noret call

As part of command module processing, it is not expected to return
and process commands when noret function for given module is called.
This patch adds wfe for noret functions of i2c slave and firmware update
module.

Change-Id: I0e42ca101318055e1e89d9634fd1c019dace4528
---

diff --git a/common/module_fwupd.c b/common/module_fwupd.c
index 4dc3d18..68f1d34 100644
--- a/common/module_fwupd.c
+++ b/common/module_fwupd.c
@@ -648,4 +648,5 @@
 void module_fwupd_run_noret(void)
 {
 	fit_run_image();
+	wfe();
 }
diff --git a/common/module_i2c_slave.c b/common/module_i2c_slave.c
index 827065e..0595c7f 100644
--- a/common/module_i2c_slave.c
+++ b/common/module_i2c_slave.c
@@ -582,4 +582,6 @@
 		i2cslave_reset_dev();
 	else
 		i2cslave_fit_run_image();
+
+	wfe();
 }
